app/controllers/items_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ItemsController < ApplicationController
+  before_action :set_item, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @items=Item.all
+  end
+
+  def show
+    @item=Item.find_by(id:params[:id])
+  end
+
+  def new
+    @item = Item.new
+  end
+
+  def create
+    @item=Item.new(item_params)
+    authorize @item
+    if @item.save
```

Comment:
I think for creating item, user just need to be sign in there is not no need of authorization here.

Reply:
According to docs only admin can add items.